cognite/client/data_classes/contextualization.py

```diff
@@ -0,0 +1,231 @@
+            time.sleep(interval)
+        if self.status == "Failed":
+            raise ModelFailedException(self.__class__.__name__, self.job_id, self.error_message)
+
+    @property
+    def result(self):
+        """Waits for the job to finish and returns the results."""
+        if not self._result:
+            self.wait_for_completion()
+        return self._result
+
+    def __str__(self):
+        return "%s(id: %d,status: %s,error: %s)" % (
+            self.__class__.__name__,
+            self.job_id,
+            self.status,
+            self.error_message,
+        )
+
+    @classmethod
+    def _load_with_status(cls, data, status_path, cognite_client):
+        obj = cls._load(data, cognite_client=cognite_client)
+        obj._status_path = status_path
+        return obj
+
+
+class EntityMatchingModel(CogniteResource):
+    _RESOURCE_PATH = "/context/entitymatching"
+    _STATUS_PATH = _RESOURCE_PATH + "/"
+
+    def __init__(
+        self,
+        id=None,
+        status=None,
+        error_message=None,
+        created_time=None,
+        start_time=None,
+        status_time=None,
+        cognite_client=None,
+        classifier=None,
+        feature_type=None,
+        match_fields=None,
+        model_type=None,
+        name=None,
+        description=None,
+        external_id=None,
+    ):
+        self.id = id
+        self.status = status
+        self.created_time = created_time
+        self.start_time = start_time
+        self.status_time = status_time
+        self.error_message = error_message
+        self.classifier = classifier
+        self.feature_type = feature_type
+        self.match_fields = match_fields
+        self.model_type = model_type
+        self.name = name
+        self.description = description
+        self.external_id = external_id
+        self._cognite_client = cognite_client
+
+    def __str__(self):
+        return "%s(id: %d,status: %s,error: %s)" % (self.__class__.__name__, self.id, self.status, self.error_message)
+
+    def update_status(self) -> str:
+        """Updates the model status and returns it"""
+        data = self._cognite_client.entity_matching._get(f"{self._STATUS_PATH}{self.id}").json()
+        self.status = data["status"]
+        self.status_time = data.get("statusTime")
+        self.start_time = data.get("startTime")
+        self.created_time = self.created_time or data.get("createdTime")
+        self.error_message = data.get("errorMessage")
+        return self.status
+
+    def wait_for_completion(self, interval=1):
+        """Waits for model completion, raising ModelFailedException if fit failed - generally not needed to call as it is called by predict"""
+        while True:
+            self.update_status()
+            if self.status not in ["Queued", "Running"]:
+                break
+            time.sleep(interval)
+        if self.status == "Failed":
+            raise ModelFailedException(self.__class__.__name__, self.id, self.error_message)
```

Comment:
:thinking: i guess you have some experience that this works ok? I can see the need for this utility, but i am not a fan of having this in the sdk :stuck_out_tongue:

Reply:
added a timeout, but other methods will still use this without one. it works very nicely in practice